test/server/options_impl_test.cc

```diff
@@ -332,45 +333,95 @@ TEST_F(OptionsImplTest, SetBothConcurrencyAndCpuset) {
 
 #if defined(__linux__)
 
+using testing::DoAll;
 using testing::Return;
+using testing::SetArgPointee;
 
 class OptionsImplPlatformLinuxTest : public testing::Test {
 public:
 };
 
 TEST_F(OptionsImplPlatformLinuxTest, AffinityTest1) {
   // Success case: cpuset size and hardware thread count are the same.
-  unsigned int fake_cpuset_size = std::thread::hardware_concurrency();
-  unsigned int fake_hw_threads = fake_cpuset_size;
+  unsigned int fake_hw_threads = 4;
+  cpu_set_t test_set;
+  Api::MockLinuxOsSysCalls linux_os_sys_calls;
+  TestThreadsafeSingletonInjector<Api::LinuxOsSysCallsImpl> linux_os_calls(&linux_os_sys_calls);
 
-  EXPECT_EQ(OptionsImplPlatformLinux::getCpuAffinityCount(fake_hw_threads), fake_cpuset_size);
+  // Set cpuset size to be four.
+  CPU_ZERO(&test_set);
+  CPU_SET(0, &test_set);
+  CPU_SET(3, &test_set);
+  CPU_SET(4, &test_set);
+  CPU_SET(7, &test_set);
+
+  EXPECT_CALL(linux_os_sys_calls, sched_getaffinity(_, _, _))
+      .WillOnce(DoAll(SetArgPointee<2>(test_set), Return(Api::SysCallIntResult{0, 0})));
+  EXPECT_EQ(OptionsImplPlatformLinux::getCpuAffinityCount(fake_hw_threads), 4);
 }
 
 TEST_F(OptionsImplPlatformLinuxTest, AffinityTest2) {
   // Success case: cpuset size is half of the hardware thread count.
-  unsigned int fake_cpuset_size = std::thread::hardware_concurrency();
-  unsigned int fake_hw_threads = 2 * fake_cpuset_size;
+  unsigned int fake_hw_threads = 16;
+  cpu_set_t test_set;
+  Api::MockLinuxOsSysCalls linux_os_sys_calls;
+  TestThreadsafeSingletonInjector<Api::LinuxOsSysCallsImpl> linux_os_calls(&linux_os_sys_calls);
+
+  // Set cpuset size to be eight.
```

Comment:
Normally I prefer to see really explicit stuff like this in tests, but in this particular case I think it might be clearer to use a loop. Or if there's some specific meaning to the non-contiguous CPU numbers you've set here it'd be worth expanding the comment to explain why that's done.

Reply:
Ok, changed to for loop. My original idea was to emphasize that the CPU ids are arbitrary and only the cpuset size matters. But I agree that it looks better this way.